src/bot/scripts/card.js

```diff
@@ -728,946 +66,1233 @@ const card = new Command({
+                  message.channel.send(`error sending card art: ${err}`);
+                });
+            }
 
-      embed.description += `Full Stats: https://sppd.feinwaru.com/${card.Image}`;
+            // friendly-fight/challenge stats
+            // ignore this if the command is -card ff,
+            // and assume 'ff' is a card name instead
+            // this needs to be called after card search
+            // as we need the rarity of the card
+
+            function _calculateCardAugmentData(original, utype, uvalue) {
+              const card = original;
+
+              const upgradeSequence = [4, 10, 10, 15, 15, 15];
+
+              const baseLevel = 1;
+              const baseUpgrade = 1;
+
+              const type = utype;
+              if (type == null) {
+                return console.error("no type specified");
+              }
+              if (type !== "u" && type !== "l") {
+                return console.error("incorrect type specified");
+              }
+
+              const value = parseInt(uvalue);
+              if (isNaN(value) === true) {
+                return console.error("no value specified");
+              }
+              if (typeof value !== "number") {
+                return console.error("incorrect value specified");
+              }
+              if (
+                value < 1 ||
+                type === "l" && value > 7 ||
+                type === "u" && value > 70
+              ) {
+                return console.error("out of bounds value specified");
+              }
+
+              let requiredLevels = 0;
+              let requiredUpgrades = 0;
+
+              if (type === "u") {
+                let currentLevels = 0;
+                let currentUpgrades = 0;
+
+                for (let i = 0; i < upgradeSequence.length; i++) {
+                  currentUpgrades += upgradeSequence[i];
+
+                  if (value < currentUpgrades + baseUpgrade + 1) {
+                    break;
+                  }
+
+                  currentLevels++;
+                }
+
+                requiredLevels = currentLevels;
+                requiredUpgrades = value - baseUpgrade;
+              }
+
+              if (type === "l") {
+                requiredLevels = value - baseLevel;
+                requiredUpgrades = upgradeSequence
+                  .slice(0, requiredLevels)
+                  .reduce((a, c) => a + c, 0);
+              }
+
+              const addReduceSlot = (a, c) =>
+                a[c.property] == null ? {
+                  ...a,
+                  [c.property]: c.value
+                } : {
+                  ...a,
+                  [c.property]: a[c.property] + c.value
+                };
+
+              const alteredStats = card.tech_tree.slots
+
+                .slice(0, requiredUpgrades)
+                .reduce(
+                  addReduceSlot,
+                  card.tech_tree.levels
+                  .slice(0, requiredLevels)
```

Comment:
these 3 lines arent properly indented

Reply:
How should they look?

Reply:
uhh how should they look?

Reply:
these should be indented once more
![image](https://user-images.githubusercontent.com/18635887/71321978-9d032480-24b9-11ea-8a90-51444e114efd.png)

Reply:
![image](https://user-images.githubusercontent.com/54243490/71322952-76e38180-24c5-11ea-9d80-c9700cca895a.png)
uhh like this?